src/Microsoft.AspNetCore.WebSockets/WebSocketMiddleware.cs

```diff
@@ -65,19 +86,26 @@ public bool IsWebSocketRequest
             {
                 get
                 {
-                    if (!_upgradeFeature.IsUpgradableRequest)
-                    {
-                        return false;
-                    }
-                    var headers = new List<KeyValuePair<string, string>>();
-                    foreach (string headerName in HandshakeHelpers.NeededHeaders)
+                    if (_isWebSocketRequest == null)
```

Comment:
Features aren't expected to be accessed concurrently, is that correct @Tratcher @davidfowl ?